server/mqtt.go

```diff
@@ -4515,10 +4619,16 @@ func (sess *mqttSession) ensurePubRelConsumerSubscription(c *client) error {
 			c.Errorf("Unable to add JetStream consumer for PUBREL for client %q: err=%v", sess.id, err)
 			return err
 		}
-		sess.pubRelConsumer = &ccr.Config
-		sess.tmaxack += maxAckPending
+		pubRelConsumer = &ccr.Config
+		tmaxack += maxAckPending
```

Comment:
We have the same "issue" in `processJSConsumer()` where in the `else` of the `if exists {` statement, we access sess.tmaxack, idHash without locking and bump tmaxack at the end also without locking (lines 4691, 4696 and 4718). Would you mind also fixing that?

Reply:
Please review https://github.com/nats-io/nats-server/pull/4760/commits/8d6a98f4ffb7bfad8dad4243c449aa6e438db0e1

Reply:
Resolving this comment since already have new threads